### PR TITLE
sturgeon: fix usb

### PR DESCRIPTION
**Pull request by timbz**
boots postmarket to debug shell

### CODE REVIEW

File: arch/arm/boot/dts/qcom-apq8026-huawei-sturgeon.dts [hunk 1]
```diff
@@ -192,6 +188,15 @@
 	pinctrl-0 = <&sdhc1_pin_a>;
 };
 
+&smbb {
+	qcom,fast-charge-safe-current = <450000>;
+	qcom,fast-charge-current-limit = <400000>;
+	qcom,fast-charge-safe-voltage = <4350000>;
+	qcom,fast-charge-high-threshold-voltage = <4350000>;
+	qcom,auto-recharge-threshold-voltage = <4240000>;
+	qcom,minimum-input-voltage = <4450000>;
```
Comment (z3ntu) [inline arch/arm/boot/dts/qcom-apq8026-huawei-sturgeon.dts]: Did you validate that these values are like in downstream?

Comment (timbz) [reply arch/arm/boot/dts/qcom-apq8026-huawei-sturgeon.dts]: I think its https://github.com/timbz/android-kernel-msm-sturgeon/blob/android-msm-sturgeon-3.10-nougat-dr1-wear/arch/arm/boot/dts/apq8026-sturgeon/apq8026-sturgeon-pm.dtsi#L84 but i am not 100% sure how they map. can take a look?

Comment (z3ntu) [reply arch/arm/boot/dts/qcom-apq8026-huawei-sturgeon.dts]: This should be the correct values
```
qcom,fast-charge-safe-voltage = <4370000>;
qcom,fast-charge-high-threshold-voltage = <4350000>;
qcom,minimum-input-voltage = <4370000>;
qcom,fast-charge-current-limit = <300000>;
qcom,fast-charge-safe-current = <600000>;
qcom,auto-recharge-threshold-voltage = <4250000>;
```

Please also post dmesg with these values, smbb prints some extra values with this branch

Comment (timbz) [reply arch/arm/boot/dts/qcom-apq8026-huawei-sturgeon.dts]: Done
```
[    0.000000] Booting Linux on physical CPU 0x0
[    0.000000] Linux version 5.16.0-rc1 (pmos@focal-fossa) (armv7-alpine-linux-musleabihf-gcc (Alpine 11.2.1_git20220117) 11.2.1 20220117, GNU ld (GNU Binutils) 2.38) #60 SMP PREEMPT Sat Feb 12 10:15:13 UTC 2022
[    0.000000] CPU: ARMv7 Processor [410fc073] revision 3 (ARMv7), cr=10c5387d
[    0.000000] CPU: div instructions available: patching division code
[    0.000000] CPU: PIPT / VIPT nonaliasing data cache, VIPT aliasing instruction cache
[    0.000000] OF: fdt: Machine model: Huawei Watch
[    0.000000] Memory policy: Data cache writealloc
[    0.000000] cma: Reserved 64 MiB at 0x1c000000
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x000000001fffffff]
[    0.000000]   HighMem  empty
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000002ffffff]
[    0.000000]   node   0: [mem 0x0000000003000000-0x00000000030fffff]
[    0.000000]   node   0: [mem 0x0000000003100000-0x000000000dbfffff]
[    0.000000]   node   0: [mem 0x000000000dc00000-0x000000000f4fffff]
[    0.000000]   node   0: [mem 0x000000000f500000-0x000000001fffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x000000001fffffff]
[    0.000000] percpu: Embedded 15 pages/cpu s32716 r8192 d20532 u61440
[    0.000000] pcpu-alloc: s32716 r8192 d20532 u61440 alloc=15*4096
[    0.000000] pcpu-alloc: [0] 0 
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 130048
[    0.000000] Kernel command line: androidboot.hardware=sturgeon user_debug=31 maxcpus=4 msm_rtb.filter=0x3F pm_levels.sleep_disabled=1 selinux=0 androidboot.emmc=true androidboot.serialno=FFS7N15909000664 androidboot.baseband=apq mdss_mdp.panel=1:dsi:0:qcom,mdss_dsi_auo_rm69080_hvga_video androidboot.emmcid=ca9a1002 androidboot.bootloader=STURGEONV4.4
[    0.000000] Unknown kernel command line parameters "user_debug=31 selinux=0", will be passed to user space.
[    0.000000] Dentry cache hash table entries: 65536 (order: 6, 262144 bytes, linear)
[    0.000000] Inode-cache hash table entries: 32768 (order: 5, 131072 bytes, linear)
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] Memory: 406896K/524288K available (11264K kernel code, 1562K rwdata, 4292K rodata, 1024K init, 257K bss, 51856K reserved, 65536K cma-reserved, 0K highmem)
[    0.000000] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=1, Nodes=1
[    0.000000] trace event string verifier disabled
[    0.000000] rcu: Preemptible hierarchical RCU implementation.
[    0.000000] rcu:     RCU event tracing is enabled.
[    0.000000] rcu:     RCU restricting CPUs from NR_CPUS=4 to nr_cpu_ids=1.
[    0.000000]  Trampoline variant of Tasks RCU enabled.
[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 10 jiffies.
[    0.000000] rcu: Adjusting geometry for rcu_fanout_leaf=16, nr_cpu_ids=1
[    0.000000] NR_IRQS: 16, nr_irqs: 16, preallocated irqs: 16
[    0.000000] random: get_random_bytes called from start_kernel+0x514/0x69c with crng_init=0
[    0.000000] arch_timer: cp15 and mmio timer(s) running at 19.20MHz (virt/virt).
[    0.000000] clocksource: arch_sys_counter: mask: 0xffffffffffffff max_cycles: 0x46d987e47, max_idle_ns: 440795202767 ns
[    0.000001] sched_clock: 56 bits at 19MHz, resolution 52ns, wraps every 4398046511078ns
[    0.000023] Switching to timer-based delay loop, resolution 52ns
[    0.000188] Console: colour dummy device 80x30
[    0.000895] printk: console [tty0] enabled
[    0.000939] Calibrating delay loop (skipped), value calculated using timer frequency.. 38.40 BogoMIPS (lpj=192000)
[    0.000984] pid_max: default: 32768 minimum: 301
[    0.001116] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.001157] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.001877] CPU: Testing write buffer coherency: ok
[    0.002124] missing device node for CPU 0
[    0.002153] CPU0: thread -1, cpu 0, socket 0, mpidr 80000000
[    0.002821] Setting up static identity map for 0x300000 - 0x300060
[    0.002960] rcu: Hierarchical SRCU implementation.
[    0.003427] smp: Bringing up secondary CPUs ...
[    0.003456] smp: Brought up 1 node, 1 CPU
[    0.003479] SMP: Total of 1 processors activated (38.40 BogoMIPS).
[    0.003505] CPU: All CPU(s) started in SVC mode.
[    0.003923] devtmpfs: initialized
[    0.008834] VFP support v0.3: implementor 41 architecture 2 part 30 variant 7 rev 3
[    0.009079] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.009127] futex hash table entries: 256 (order: 2, 16384 bytes, linear)
[    0.012106] pinctrl core: initialized pinctrl subsystem
[    0.013325] NET: Registered PF_NETLINK/PF_ROUTE protocol family
[    0.014890] DMA: preallocated 256 KiB pool for atomic coherent allocations
[    0.016279] thermal_sys: Registered thermal governor 'step_wise'
[    0.016420] cpuidle: using governor menu
[    0.016598] NET: Registered PF_QIPCRTR protocol family
[    0.016812] hw-breakpoint: found 5 (+1 reserved) breakpoint and 4 watchpoint registers.
[    0.016851] hw-breakpoint: maximum watchpoint size is 8 bytes.
[    0.040647] kprobes: kprobe jump-optimization is enabled. All kprobes are optimized if possible.
[    0.042158] iommu: Default domain type: Translated 
[    0.042189] iommu: DMA domain TLB invalidation policy: strict mode 
[    0.042418] vgaarb: loaded
[    0.042852] SCSI subsystem initialized
[    0.043061] libata version 3.00 loaded.
[    0.043326] usbcore: registered new interface driver usbfs
[    0.043406] usbcore: registered new interface driver hub
[    0.043470] usbcore: registered new device driver usb
[    0.043612] mc: Linux media interface: v0.10
[    0.043668] videodev: Linux video capture interface: v2.00
[    0.044007] qcom_scm: convention: smc legacy
[    0.044501] Advanced Linux Sound Architecture Driver Initialized.
[    0.045086] Bluetooth: Core ver 2.22
[    0.045148] NET: Registered PF_BLUETOOTH protocol family
[    0.045172] Bluetooth: HCI device and connection manager initialized
[    0.045203] Bluetooth: HCI socket layer initialized
[    0.045230] Bluetooth: L2CAP socket layer initialized
[    0.045280] Bluetooth: SCO socket layer initialized
[    0.045847] clocksource: Switched to clocksource arch_sys_counter
[    0.132328] NET: Registered PF_INET protocol family
[    0.132552] IP idents hash table entries: 8192 (order: 4, 65536 bytes, linear)
[    0.133391] tcp_listen_portaddr_hash hash table entries: 512 (order: 0, 6144 bytes, linear)
[    0.133457] TCP established hash table entries: 4096 (order: 2, 16384 bytes, linear)
[    0.133537] TCP bind hash table entries: 4096 (order: 3, 32768 bytes, linear)
[    0.133635] TCP: Hash tables configured (established 4096 bind 4096)
[    0.133737] UDP hash table entries: 256 (order: 1, 8192 bytes, linear)
[    0.133791] UDP-Lite hash table entries: 256 (order: 1, 8192 bytes, linear)
[    0.133984] NET: Registered PF_UNIX/PF_LOCAL protocol family
[    0.134470] RPC: Registered named UNIX socket transport module.
[    0.134505] RPC: Registered udp transport module.
[    0.134526] RPC: Registered tcp transport module.
[    0.134545] RPC: Registered tcp NFSv4.1 backchannel transport module.
[    0.145881] PCI: CLS 0 bytes, default 64
[    0.147039] Initialise system trusted keyrings
[    0.147305] workingset: timestamp_bits=30 max_order=17 bucket_order=0
[    0.153057] Trying to unpack rootfs image as initramfs...
[    0.166400] NFS: Registering the id_resolver key type
[    0.166463] Key type id_resolver registered
[    0.166485] Key type id_legacy registered
[    0.176279] Key type cifs.idmap registered
[    0.176319] jffs2: version 2.2. (NAND) © 2001-2006 Red Hat, Inc.
[    0.176669] fuse: init (API version 7.35)
[    0.338928] Freeing initrd memory: 1220K
[    0.458929] Key type asymmetric registered
[    0.458959] Asymmetric key parser 'x509' registered
[    0.459166] Block layer SCSI generic (bsg) driver version 0.4 loaded (major 248)
[    0.459201] io scheduler mq-deadline registered
[    0.459223] io scheduler kyber registered
[    0.477575] msm_serial: driver initialized
[    0.497950] brd: module loaded
[    0.509237] loop: module loaded
[    0.549771] SCSI Media Changer driver v0.25 
[    0.550563] spmi spmi-0: PMIC arbiter version v1 (0x20000002)
[    0.554525] libphy: Fixed MDIO Bus: probed
[    0.555274] SLIP: version 0.8.4-NET3.019-NEWTTY (dynamic channels, max=256) (6 bit encapsulation enabled).
[    0.555312] CSLIP: code copyright 1989 Regents of the University of California.
[    0.555383] usbcore: registered new interface driver ax88179_178a
[    0.555447] usbcore: registered new interface driver cdc_ether
[    0.555508] usbcore: registered new interface driver net1080
[    0.555568] usbcore: registered new interface driver cdc_subset
[    0.555648] usbcore: registered new interface driver cdc_ncm
[    0.566240] ehci_hcd: USB 2.0 'Enhanced' Host Controller (EHCI) Driver
[    0.566271] ehci-pci: EHCI PCI platform driver
[    0.566388] usbcore: registered new interface driver cdc_acm
[    0.566416] cdc_acm: USB Abstract Control Model driver for USB modems and ISDN adapters
[    0.567195] UDC core: couldn't find an available UDC - added [g_ether] to list of pending drivers
[    0.567948] input: pm8941_pwrkey as /devices/platform/soc/fc4cf000.spmi/spmi-0/0-00/fc4cf000.spmi:pm8226@0:pwrkey@800/input/input0
[    0.569099] i2c_dev: i2c /dev entries driver
[    0.569320] i2c_qup f9927000.i2c: 
[    0.569320]  tx channel not available
[    0.570547] qcom-smbb fc4cf000.spmi:pm8226@0:charger@1000: Initializing SMBB rev 1
[    0.570604] qcom-smbb fc4cf000.spmi:pm8226@0:charger@1000: qcom,fast-charge-safe-current <= 600000
[    0.570638] qcom-smbb fc4cf000.spmi:pm8226@0:charger@1000: qcom,fast-charge-current-limit <= 300000
[    0.570660] qcom-smbb fc4cf000.spmi:pm8226@0:charger@1000: usb-charge-current-limit => 1500000
[    0.570682] qcom-smbb fc4cf000.spmi:pm8226@0:charger@1000: qcom,dc-current-limit => 1500000
[    0.570706] qcom-smbb fc4cf000.spmi:pm8226@0:charger@1000: qcom,fast-charge-safe-voltage <= 4370000
[    0.570738] qcom-smbb fc4cf000.spmi:pm8226@0:charger@1000: qcom,fast-charge-high-threshold-voltage <= 4350000
[    0.570763] qcom-smbb fc4cf000.spmi:pm8226@0:charger@1000: qcom,fast-charge-low-threshold-voltage => 3200000
[    0.570786] qcom-smbb fc4cf000.spmi:pm8226@0:charger@1000: qcom,auto-recharge-threshold-voltage inaccurate, rounded to 4240000
[    0.570827] qcom-smbb fc4cf000.spmi:pm8226@0:charger@1000: qcom,auto-recharge-threshold-voltage <= 4240000
[    0.570850] qcom-smbb fc4cf000.spmi:pm8226@0:charger@1000: qcom,minimum-input-voltage inaccurate, rounded to 4350000
[    0.570886] qcom-smbb fc4cf000.spmi:pm8226@0:charger@1000: qcom,minimum-input-voltage <= 4350000
[    0.572120] qcom-smbb fc4cf000.spmi:pm8226@0:charger@1000: status = 000
[    0.576104] qcom-smbb fc4cf000.spmi:pm8226@0:charger@1000: status = 000
[    0.576401] qcom-smbb fc4cf000.spmi:pm8226@0:charger@1000: status = 000
[    0.586322] qcom-smbb fc4cf000.spmi:pm8226@0:charger@1000: status = 018
[    0.586640] qcom-smbb fc4cf000.spmi:pm8226@0:charger@1000: status = 018
[    0.586962] qcom-smbb fc4cf000.spmi:pm8226@0:charger@1000: status = 019
[    0.587379] qcom-smbb fc4cf000.spmi:pm8226@0:charger@1000: status = 019
[    0.588781] device-mapper: ioctl: 4.45.0-ioctl (2021-03-22) initialised: dm-devel@redhat.com
[    0.589079] sdhci: Secure Digital Host Controller Interface driver
[    0.589106] sdhci: Copyright(c) Pierre Ossman
[    0.589125] sdhci-pltfm: SDHCI platform and OF driver helper
[    0.589613] usbcore: registered new interface driver usbhid
[    0.589642] usbhid: USB HID core driver
[    0.590721] SPI driver bmp280 has no spi_device_id for bosch,bmp085
[    0.591863] NET: Registered PF_PACKET protocol family
[    0.591978] Key type dns_resolver registered
[    0.592154] Registering SWP/SWPB emulation handler
[    0.592380] Loading compiled-in X.509 certificates
[    0.737227] qcom_q6v5_pas adsp-pil: deferred probe timeout, ignoring dependency
[    0.737283] qcom_q6v5_pas: probe of adsp-pil failed with error -110
[    0.739022] cfg80211: Loading compiled-in X.509 certificates for regulatory database
[    0.742358] s1: Bringing 0uV into 500000-500000uV
[    0.743020] s3: Bringing 0uV into 1200000-1200000uV
[    0.743369] s4: Bringing 0uV into 1800000-1800000uV
[    0.743732] s5: Bringing 0uV into 1150000-1150000uV
[    0.744084] l1: Bringing 0uV into 1225000-1225000uV
[    0.744437] l2: Bringing 0uV into 1200000-1200000uV
[    0.744785] l3: Bringing 0uV into 750000-750000uV
[    0.745145] l4: Bringing 0uV into 1200000-1200000uV
[    0.745510] l5: Bringing 0uV into 1200000-1200000uV
[    0.758250] cfg80211: Loaded X.509 cert 'sforshee: 00b28ddf47aef9cea7'
[    0.758430] l6: Bringing 0uV into 1800000-1800000uV
[    0.758826] l7: Bringing 0uV into 1850000-1850000uV
[    0.759199] l8: Bringing 0uV into 1800000-1800000uV
[    0.759570] l9: Bringing 0uV into 2050000-2050000uV
[    0.759957] l10: Bringing 0uV into 1800000-1800000uV
[    0.760630] l12: Bringing 0uV into 1800000-1800000uV
[    0.761307] l14: Bringing 0uV into 2750000-2750000uV
[    0.761693] l15: Bringing 0uV into 2800000-2800000uV
[    0.762073] l16: Bringing 0uV into 3000000-3000000uV
[    0.762470] l17: Bringing 0uV into 2950000-2950000uV
[    0.762866] l18: Bringing 0uV into 2950000-2950000uV
[    0.763263] l19: Bringing 0uV into 2850000-2850000uV
[    0.763659] l20: Bringing 0uV into 3075000-3075000uV
[    0.764068] l21: Bringing 0uV into 1800000-1800000uV
[    0.764471] l22: Bringing 0uV into 1800000-1800000uV
[    0.764874] l23: Bringing 0uV into 1800000-1800000uV
[    0.765278] l24: Bringing 0uV into 1300000-1300000uV
[    0.765687] l25: Bringing 0uV into 1775000-1775000uV
[    0.776147] l26: Bringing 0uV into 1225000-1225000uV
[    0.776558] l27: Bringing 0uV into 2050000-2050000uV
[    0.776962] l28: Bringing 0uV into 1800000-1800000uV
[    0.779009] platform regulatory.0: Direct firmware load for regulatory.db failed with error -2
[    0.779059] cfg80211: failed to load regulatory.db
[    0.795958] rmi4_i2c 0-0020: rmi_set_page: set page failed: -6.
[    0.795992] rmi4_i2c 0-0020: Failed to set page select to 0
[    0.807474] using random self ethernet address
[    0.807507] using random host ethernet address
[    0.808055] usb0: HOST MAC 16:e4:66:c0:e7:18
[    0.808115] usb0: MAC 02:90:c3:29:72:fd
[    0.808158] using random self ethernet address
[    0.808183] using random host ethernet address
[    0.808276] g_ether gadget: Ethernet Gadget, version: Memorial Day 2008
[    0.808306] g_ether gadget: g_ether ready
[    0.808747] l20: voltage operation not allowed
[    0.845873] mmc0: SDHCI controller on f9824900.sdhci [f9824900.sdhci] using ADMA
[    0.950871] mmc0: new HS200 MMC card at address 0001
[    0.956184] mmcblk0: mmc0:0001 H4G2a 3.64 GiB 
[    0.959660]  mmcblk0: p1 p2 p3 p4 p5 p6 p7 p8 p9 p10 p11 p12 p13 p14 p15 p16 p17 p18 p19 p20 p21 p22 p23 p24 p25 p26
[    0.964126] mmcblk0boot0: mmc0:0001 H4G2a 4.00 MiB 
[    0.967420] mmcblk0boot1: mmc0:0001 H4G2a 4.00 MiB 
[    0.977232] mmcblk0rpmb: mmc0:0001 H4G2a 4.00 MiB, chardev (245:0)
[    1.230005] ALSA device list:
[    1.230041]   No soundcards found.
[    1.231889] Freeing unused kernel image (initmem) memory: 1024K
[    1.256163] Run /init as init process
[    1.256190]   with arguments:
[    1.256196]     /init
[    1.256202]   with environment:
[    1.256207]     HOME=/
[    1.256214]     TERM=linux
[    1.256219]     user_debug=31
[    1.256225]     selinux=0
[    1.494360] random: fast init done
[    2.029892] qcom-smbb fc4cf000.spmi:pm8226@0:charger@1000: status = 099
[   23.693798] using random self ethernet address
[   23.693848] using random host ethernet address
[   23.706984] UDC core: couldn't find an available UDC or it's busy: -16
```

Comment (z3ntu) [reply arch/arm/boot/dts/qcom-apq8026-huawei-sturgeon.dts]: Looks fine, but please take in these two updated values
```
qcom,auto-recharge-threshold-voltage = <4240000>;
qcom,minimum-input-voltage = <4350000>;
```
You can see in the dmesg the driver rounded them so let's just provide it with already-rounded values

Comment (timbz) [reply arch/arm/boot/dts/qcom-apq8026-huawei-sturgeon.dts]: done

**Comment by z3ntu** [comment]
Thank you!

**Pull request merged**